homeassistant/components/recorder/statistics.py

```diff
@@ -97,16 +100,40 @@ def statistics_during_period(hass, start_time, end_time=None, statistic_id=None)
 
         statistic_ids = [statistic_id] if statistic_id is not None else None
 
-        return _sorted_statistics_to_dict(
-            hass, session, stats, start_time, statistic_ids
+        return _sorted_statistics_to_dict(stats, statistic_ids)
+
+
+def get_last_statistics(hass, number_of_stats, statistic_id=None):
+    """Return the last number_of_stats statistics."""
+    with session_scope(hass=hass) as session:
+        baked_query = hass.data[STATISTICS_BAKERY](
+            lambda session: session.query(*QUERY_STATISTICS)
+        )
+
+        if statistic_id is not None:
+            baked_query += lambda q: q.filter_by(statistic_id=bindparam("statistic_id"))
+            statistic_id = statistic_id.lower()
```

Comment:
Why would we lower it. Better to require the caller to pass in the right one

Reply:
```suggestion
```

Reply:
Fixed in7f48705